### PR TITLE
Report RTMP client connection error

### PR DESCRIPTION
**Pull request by harryz2000**

### CODE REVIEW

File: src/RTMPClientConnectionImpl.i [hunk 1]
```diff
@@ -33,7 +33,13 @@ public:
 	{
 		UTF8Parser parser;
 		parser.SetString(*Nan::Utf8String(url));
-		RTMPClientConnection::SendCommand(id, L"publish", nullptr, new AMFString(parser.GetWChar()));
+		RTMPClientConnection::SendCommand(id, L"publish", nullptr, new AMFString(parser.GetWChar()), 
+			[=](bool isError, AMFData* params, const std::vector<AMFData*>& extra) {
+				if (isError)
+				{
+					onError(this, ErrorCode::PublishCommandFailed);
```
Comment (murillo128) [inline src/RTMPClientConnectionImpl.i:40]: we should not send a client wide error if the `Publish` cmd has an error. Let's make the Publish take a promise as input as  CreateStream/DeleteStream instead.

File: lib/ClientConnection.js [hunk 2]
```diff
@@ -72,6 +73,19 @@ class ClientConnection extends Emitter
 			//Stop us
 			this.stop();
 		};
+		
+		this.onerror = (/** @type {Number} */errorCode) =>
```
Comment (murillo128) [inline lib/ClientConnection.js:77]: should we have a different error handler or just add the reason code to the stopped event as in websocket?

https://developer.mozilla.org/en-US/docs/Web/API/CloseEvent/code

@mildsunrise wdyt?

**Review by murillo128 — CHANGES_REQUESTED** [review]

File: src/RTMPClientConnectionImpl.i (hunk 1 above)
Comment (harryz2000) [reply src/RTMPClientConnectionImpl.i:40]: Yep. Will do.

File: lib/ClientConnection.js (hunk 2 above)
Comment (harryz2000) [reply lib/ClientConnection.js:77]: I assume you meant combine ondisconnect and onerror together? what do you mean the stopped event? The ResourceEntry.stop()?

File: src/RTMPClientConnectionImpl.i (hunk 1 above)
Comment (harryz2000) [reply src/RTMPClientConnectionImpl.i:40]: It appears the server wouldn't repond _error/_result. The server would return onStatus instead. Therefore, this callback wouldn't be called. I would revert this change.

> "The server responds with the onStatus command to mark the beginning of publish."

**Review by murillo128 — APPROVED** [review]
lgtm

File: lib/ClientConnection.js (hunk 2 above)
Comment (mildsunrise) [reply lib/ClientConnection.js]: we can do as websocket then, yeah

**Review by murillo128 — CHANGES_REQUESTED** [review]
please add tests

**Review by murillo128 — APPROVED** [review]
lgtm

**Pull request merged**